Src/main.c

```diff
@@ -103,13 +103,37 @@ int main(void)
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
+  //Initialize ADC and TIM
+  HAL_ADC_Init(&hadc);
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  //set servo PWM output relative to input voltage from potentiometer.
+  //potentiometer voltage from 0-3.3V, PWM modulation from 1-2ms on-time at 50Hz (period = 20ms)
+  
+  int Ch1 = 0; //no need to constantly re-initialize a variable.
   while (1)
   {
+      Ch1 = 0;
+      //ADC conversion:
+      double OnTime = HAL_ADC_GetValue(&hadc); //RAW ADC input
+      OnTime = OnTime / (1023 / 3.3); //Converting raw to input voltage value for clarity (between 0V and 3.3V)
+      OnTime = (OnTime / 3.3) + 1; //Converting to On-time of the PWM by getting the decimal of input Voltage and adding 1ms 
```

Comment:
This part doesn't make too much sense to me - Say the pot value is 1023 (aka 3V3, and max servo on is 2ms)
In line 121, OnTime = (1023)/(1023*3.3) = 0.303030303
In line 122, OnTime = 0.303030303/3.3 +1 = 1.091

Could you re-explain your logic with this?

Reply:
Also - use const's to make your code easier to read!